### PR TITLE
OPL-9: Added Invoice Payments

### PR DESCRIPTION
**Pull request by dborowiecki**
PR include changes from [PR OPL-15](https://github.com/openimis/openimis-be-invoice_py/pull/4), please check it first.
TICKET: [OPL-9](https://openimis.atlassian.net/browse/OPL-9)

### CODE REVIEW

File: invoice/services/invoicePayments.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                invoice_payment.invoice.save(username=self.user.username)
+                return self.save_instance(invoice_payment)
+        except Exception as exc:
+            return _output_exception(model_name="InvoicePayment", method="payment_received", exception=exc)
+
+    def payment_refunded(self, invoice_payment):
+        try:
+            with transaction.atomic():
+                self.validation_class.validate_refund_payment(self.user, invoice_payment)
+                # TODO: Currently Payment model doesn't include cancelled status. Should it be added?
```
Comment (delcroip) [inline invoice/services/invoicePayments.py:52]: yes, I will update the design

File: invoice/services/invoicePayments.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                self._update_payment_status(invoice_payment, InvoicePayment.InvoicePaymentStatus.REFUNDED)
+                self._update_invoice_status(invoice_payment.invoice, Invoice.InvoiceStatus.CANCELLED)
```
Comment (delcroip) [inline invoice/services/invoicePayments.py:54]: I am not sure the invoice should become "CANCELED" but just make a SUSPENDED (like the policies)

File: invoice/services/invoicePayments.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        except Exception as exc:
+            return _output_exception(model_name="InvoicePayment", method="payment_refunded", exception=exc)
+
+    def payment_cancelled(self, invoice_payment):
+        try:
+            with transaction.atomic():
+                self.validation_class.validate_cancel_payment(self.user, invoice_payment)
+
+                # TODO: Currently Payment model doesn't include cancelled status. Should it be added?
+                self._update_invoice_status(invoice_payment.invoice, Invoice.InvoiceStatus.CANCELLED)
```
Comment (delcroip) [inline invoice/services/invoicePayments.py:67]: Same here

**Review by delcroip — CHANGES_REQUESTED** [review]
See comments

File: invoice/services/invoicePayments.py (hunk 1 above)
Comment (dborowiecki) [reply invoice/services/invoicePayments.py]: New statuses included in model.

File: invoice/services/invoicePayments.py (hunk 3 above)
Comment (dborowiecki) [reply invoice/services/invoicePayments.py]: Payment status is now changed to cancelled, invoice is suspended.

**Review by delcroip — APPROVED** [review]

**Pull request merged**